// react-native-paper-dates/lib/module/Date/Day.js
"use strict";

import { Text, TouchableRipple } from 'react-native-paper';
import { StyleSheet, View } from 'react-native';
import DayRange from './DayRange';
import { daySize } from './dateUtils';
import { memo, useCallback } from 'react';
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
function EmptyDayPure() {
  return /*#__PURE__*/_jsx(View, {
    style: styles.empty
  });
}
export const EmptyDay = /*#__PURE__*/memo(EmptyDayPure);
function Day(props) {
  const {
    day,
    month,
    year,
    selected,
    inRange,
    leftCrop,
    rightCrop,
    onPressDate,
    primaryColor,
    selectColor,
    isToday,
    disabled,
    theme
  } = props;
  const borderColor = theme.colors.primary;
  const onPress = useCallback(() => {
    onPressDate(new Date(year, month, day));
  }, [onPressDate, year, month, day]);

  // Determine text colors for M3
  let baseTextColor;
  let finalTextColor;
  if (selected) {
    baseTextColor = theme.colors.onPrimary;
  } else if (inRange && theme.dark) {
    baseTextColor = theme.colors.onPrimaryContainer;
  } else {
    baseTextColor = theme.colors.onSurface;
  }
  if (isToday) {
    finalTextColor = selected ? baseTextColor : theme.colors.primary;
  } else {
    finalTextColor = baseTextColor;
  }
  const textFont = theme.fonts.bodySmall;
  return /*#__PURE__*/_jsxs(View, {
    style: [styles.root, disabled && styles.disabled],
    children: [/*#__PURE__*/_jsx(DayRange, {
      inRange: inRange,
      leftCrop: leftCrop,
      rightCrop: rightCrop,
      selectColor: selectColor
    }), /*#__PURE__*/_jsx(TouchableRipple, {
      testID: `react-native-paper-dates-day-${year}-${month}-${day}`,
      disabled: disabled,
      borderless: true,
      onPress: disabled ? undefined : onPress,
      style: [styles.button, {
        backgroundColor: inRange ? selectColor : undefined
      }],
      accessibilityRole: "button",
      children: /*#__PURE__*/_jsx(View, {
        style: [styles.day, isToday ? {
          borderColor: borderColor
        } : null, selected ? {
          backgroundColor: primaryColor
        } : null],
        children: /*#__PURE__*/_jsx(Text, {
          maxFontSizeMultiplier: 1.5,
          style: [baseTextColor ? {
            color: finalTextColor
          } : undefined, {
            ...textFont
          }],
          selectable: false,
          children: day
        })
      })
    })]
  });
}
const styles = StyleSheet.create({
  button: {
    width: daySize,
    height: daySize,
    overflow: 'hidden',
    borderRadius: daySize / 2
  },
  day: {
    flexBasis: 0,
    flex: 1,
    borderRadius: daySize / 2,
    width: daySize,
    height: daySize,
    justifyContent: 'center',
    alignItems: 'center',
    borderWidth: 1,
    borderColor: 'transparent'
  },
  disabled: {
    opacity: 0.3
  },
  empty: {
    flex: 1,
    flexBasis: 0
  },
  root: {
    flexBasis: 0,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    position: 'relative'
  }
});
export default /*#__PURE__*/memo(Day);
